src/client.js

```diff
@@ -90,11 +89,13 @@ class MistralClient {
 
     for (let attempts = 0; attempts < this.maxRetries; attempts++) {
       try {
-        const response = await fetch(url, options);
+        const response = await this._fetch(url, options);
 
         if (response.ok) {
           if (request?.stream) {
-            if (isNode) {
+            if (isNode && !haveNativeFetch ||
+              // The test mocks do not return a body with getReader
+              typeof response.body.getReader === 'undefined') {
```

Comment:
I think here we can remove the condition on `isNode` and `haveNativeFetch` as the second part of the condition will catch this anyway